### PR TITLE
std.complex: Add complex norm() and proj()

### PR DESCRIPTION
**Pull request by ibuclaw**
A couple more trivial additions to std.complex, all that remain (future PRs) are a(sin|cos|tan), (sin|cos|tan)h, and possibly isClose.

N.B: should norm be named toNorm?

### CODE REVIEW

**Comment by ibuclaw** [comment]
@n8sh - I take that to mean the current names are fine then. :-)

File: std/complex.d [hunk 1]
```diff
@@ -780,6 +780,29 @@ T arg(T)(Complex!T z) @safe pure nothrow @nogc
...
+}
+
+///
+@safe pure nothrow @nogc unittest
+{
+    import std.math : approxEqual, PI;
+    assert(norm(complex(3.0, 4.0)) == 25.0);
+    assert(norm(fromPolar(5.0, 0.0)) == 25.0);
+    assert(approxEqual(norm(fromPolar(5.0L, PI / 6)), 25.0L));
+    assert(approxEqual(norm(fromPolar(5.0L, 13 * PI / 6)), 25.0L));
```
Comment (ibuclaw) [inline std/complex.d:802]: This could wait until #7482 is in first, then I'll switch it over to `isClose`.

**Comment by ibuclaw** [comment]
@thewilsonator - all fine here?  Also, `norm` or `toNorm` ?

**Comment by thewilsonator** [comment]
`norm` is my choice.

**Comment by ibuclaw** [comment]
> `norm` is my choice.

Ready to merge then.

**Pull request merged**